src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java

```diff
@@ -162,24 +191,6 @@ private SubmissionHistory tryGetSubmissionHistory(@NotNull Exercise exercise,
     }
   }
 
-  @CalledWithReadLock
-  @Nullable
-  private List<Path> tryGetFilePaths(@NotNull List<String> filenames,
```

Comment:
please check L76 and L86 - one call to `e.getProject()` can be "optimized" :wink: